core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceGroupV1ToInstanceGroupV4Converter.java

```diff
@@ -67,6 +73,11 @@ private InstanceGroupV1Request convert(InstanceGroupV4Request source) {
         response.setRecipeNames(source.getRecipeNames());
         response.setAws(getIfNotNull(source.getAws(), instanceGroupParameterConverter::convert));
         response.setAzure(getIfNotNull(source.getAzure(), instanceGroupParameterConverter::convert));
+        if (source.getScalingMode() == null) {
+            response.setScalingMode(ScalingMode.UNSPECIFIED);
+        } else {
+            response.setScalingMode(source.getScalingMode());
+        }
```

Comment:
this logic looks duplicated to me, could you extract it to a method?

Reply:
The types are different, it gets more complicated to de-duplicate. Instead removed that logic!